pkg/executor/index_lookup_hash_join.go

```diff
@@ -217,64 +222,44 @@ func (e *IndexNestedLoopHashJoin) wait4JoinWorkers() {
 // Next implements the IndexNestedLoopHashJoin Executor interface.
 func (e *IndexNestedLoopHashJoin) Next(ctx context.Context, req *chunk.Chunk) error {
 	if !e.prepared {
-		e.startWorkers(ctx)
+		ctx = e.startWorkers(ctx)
```

Comment:
What happend if there is a panic during the second Next() call?   The ctx will not be set during the seond call. So e.runInOrder/e.runUnordered uses the different ctx between the first call and the second call.